Content.Shared/Storage/EntitySystems/MagnetPickupSystem.cs

```diff
@@ -48,17 +52,21 @@ public override void Update(float frameTime)
 
             comp.NextScan += ScanDelay;
 
-            if (!_inventory.TryGetContainingSlot((uid, xform, meta), out var slotDef))
-                continue;
+            var parentUid = xform.ParentUid;
 
-            if ((slotDef.SlotFlags & comp.SlotFlags) == 0x0)
-                continue;
+            if (!_entityManager.TryGetComponent<BorgSwitchableTypeComponent>(parentUid, out _) || !_hands.TryGetActiveItem(parentUid, out var activeItem) && activeItem != uid)
```

Comment:
If you don't need "out" variable, use ```HasComp<AnyComponent>(uid)``` instead.

```suggestion
            if (!HasComp<BorgSwitchableTypeComponent>(parentUid) || !_hands.TryGetActiveItem(parentUid, out var activeItem) && activeItem != uid)
```

After this don't forget to remove EntityManager from dependency list (it'll be not used in this case)